app/models/katello/concerns/hostgroup_extensions.rb

```diff
@@ -95,22 +96,31 @@ def equivalent_kickstart_repository
         ks_repos.find { |repo| repo[:name] == kickstart_repository.label }
       end
 
-      def matching_kickstart_repository?
+      def matching_kickstart_repository?(content_facet)
         return true unless operatingsystem
 
         if operatingsystem.respond_to? :kickstart_repos
-          return operatingsystem.kickstart_repos(self).any? { |repo| repo[:id] == kickstart_repository_id }
+          return operatingsystem.kickstart_repos(self).any? { |repo| repo[:id] == (content_facet&.kickstart_repository_id || content_facet&.kickstart_repository&.id) }
         end
       end
 
       private
 
-      def inherited_ancestry_attribute(attribute)
-        if ancestry.present?
-          self[attribute] || self.class.sort_by_ancestry(ancestors.where("#{attribute.to_s} is not NULL")).last.try(attribute)
-        else
-          self.send(attribute)
+      def inherited_ancestry_attribute(attribute, facet)
+        value = self.send(facet)&.send(attribute)
+
+        if value.nil? && ancestry.present?
+          # value = self.class.sort_by_ancestry(ancestors.where("#{attribute.to_s} is not NULL")).last.try(attribute)
```

Comment:
Maybe this comment can be replaced with the goal of this block of code? :)

Reply:
@jturel, fixed the comment